_config/email.yml

```diff
@@ -1,12 +1,55 @@
 ---
-Name: emailconfig
+Name: mailer
 ---
 SilverStripe\Core\Injector\Injector:
-  Swift_Transport: Swift_MailTransport
-  Swift_Mailer:
-    constructor:
-      - '%$Swift_Transport'
-  SilverStripe\Control\Email\Mailer:
-    class: SilverStripe\Control\Email\SwiftMailer
-    properties:
-      SwiftMailer: '%$Swift_Mailer'
+  Symfony\Component\Mailer\MailerInterface:
+    class: Symfony\Component\Mailer\Mailer
+    constructor:
+      0: '%$Symfony\Component\Mailer\Transport\TransportInterface'
+      1: '%$Symfony\Component\Messenger\MessageBusInterface.mailer'
+  Symfony\Component\Messenger\MessageBusInterface.mailer:
```

Comment:
Why are we adding a `.mailer` to the end of all these? Is there some clash with other configuration that uses the same classes and interfaces?
And if we're using it on these, why not on `Symfony\Component\Mailer\Transport\TransportInterface`?

Reply:
There isn't currently a clash, though there may be one in the future so I think it's best to give them their own 'namespace' now

TransportInterface does not have one because it already has \Mailer\ in its identifier - there's no conceivable scenario where we end up using a 2nd Mailer

After refactoring out the bus, the only remaining identifier with the .mailer suffix is Symfony\Component\EventDispatcher\EventDispatcherInterface.mailer. It has the suffix because there's a reasonable chance that we'll use the EventDispatcherInterface for something else

This is the same reason why I changed the `name` of the yml file from `emailconfig` to `mailer` - it's so that it's all consistent with symfony/**mailer**.  In our own down we refer people to the symfony/mailer docs, we use the MAILER_DSN env variable - so I think it's correct to use the word 'mailer' over 'email' since we're so heavily leveraging symfony/mailer for email

I've just renamed the file from email.yml to mailer.yml for consistency

Reply:
I'd prefer to not add a namespace, and just add a namespace to any new config that would otherwise clash (especially since, as you point out, "there's no conceivable scenario where we end up using a 2nd Mailer") - but it's effectively an aesthetic choice so I won't hold up the PR for it. We can leave it as is.